pkg/util/ingress/ingress.go

```diff
@@ -289,12 +289,12 @@ func checkPath(paths []networkingv1.HTTPIngressPath, enablePathTypeValidation bo
 
 		case networkingv1.PathTypeExact, networkingv1.PathTypePrefix:
 			if enablePathTypeValidation {
-				if !pathAlphaNumericRegex(path.Path) {
+				if !regexSpecificChars.MatchString(path.Path) {
```

Comment:
nope, this is wrong :)

if pathTypeValidation is enabled, you should check if pathAlphaNumeric is matched, otherwise it should fail